Prox/Prox/Models/Place.swift

```diff
@@ -451,6 +451,37 @@ struct OpenHours {
         return nil
     }
 
+    func getEventTimeText(forTime baseTime: Date) -> (String, String)? {
+        // Note: this function is basically a modified version of nextOpeningTime()
+        let midnight = DateComponents(hour: 0, minute: 0)
+
+        // HACK: Keeping this as a range so we can easily change this function look ahead more days if needed.
+        // (1...N) will look ahead N days after today.
+        let times = [baseTime] + (1...1).map { getDate(forTime: midnight, onDate: baseTime.addingTimeInterval(TimeInterval($0) * AppConstants.ONE_DAY))! }
+
+        for time in times {
+            guard let allOpeningTimes = getOpeningTimes(forDate: time),
+                  let openingPeriod = getCurrentOpeningPeriod(fromOpeningPeriods: allOpeningTimes, forDate: time) ?? getNextOpeningPeriod(fromOpeningPeriods: allOpeningTimes, forDate: time) else {
+                continue
+            }
+
+            let openingDayText: String
+            if OpenHours.calendar.isDate(time, inSameDayAs: baseTime) {
```

Comment:
This is always true for index 0, the next if is always true for index 1, and else will be called for the rest – this makes the code smell to me.

That being said, in this implementation, there is a separation from the array ordering and the return value (but this abstraction adds complexity and confuses the reader about motivations! :) ).

I think there is a clearer solution here but I can't think of it off the top of my head so this is probably good enough. Maybe add a comment to say it can be clearer?